v3/src/components/case-card/case-card-model.ts

```diff
@@ -45,6 +46,46 @@ export const CaseCardModel = TileContentModel
         self.attributeColumnWidths.delete(collectionId)
       }
     },
+    addNewCase(cases: IGroupedCase[], collection: ICollectionModel, displayedCaseId: string) {
+      const newCase = {} as ICase
+  
+      // This function gets values for the given ancestor or descendant collection's attributes, and
+      // sets them on the new case
+      const setHierarchyValues = (relativeCollection: ICollectionModel, alreadyProcessed = new Set<string>()) => {
+        if (alreadyProcessed.has(relativeCollection.id)) return
+
+        alreadyProcessed.add(relativeCollection.id)
+        relativeCollection.attributes.forEach(attr => {
+          if (attr?.id) {
+            const value = self.data?.getValue(displayedCaseId, attr.id)
+            newCase[attr.id] = value
+          }
+        })
+
+        // Recursively set any parent or child collection attribute values on the new case
+        relativeCollection.parent && setHierarchyValues(relativeCollection.parent, alreadyProcessed)
+        relativeCollection.child && setHierarchyValues(relativeCollection.child, alreadyProcessed)
+      }
+    
+      // Set any parent and child collection attribute values on the new case
+      collection.parent && setHierarchyValues(collection.parent)
+      collection.child && setHierarchyValues(collection.child)
+
+      // Add empty values for the current collection's attributes since these are what the user will want to customize.
+      // Not setting them to "" results in "undefined" being displayed in the case card UI's input fields.
+      collection?.attributes.forEach(attr => {
+        if (attr?.id) {
+          newCase[attr.id] = ""
+        }
+      })
+
+      const newCases = [newCase]
+      const lastExistingCaseId = cases[cases.length - 1].__id__
+      const newCaseId = self.data?.addCases(newCases, {after: lastExistingCaseId})[0]
```

Comment:
Forgot to mention this earlier, but this can be simplified to:
```suggestion
      const [newCaseId] = self.data?.addCases(newCases) ?? []
```
because adding at the end is the default.